### DISCUSSION TITLE
Executing a command using rules after a system boot

### DISCUSSION THREAD

**Discussion by tesnaa** [arendst/Tasmota#12266] [Q&A] [answered]
Hi, 

I have Zmai-90 https://templates.blakadder.com/ZMAi-90.html and so far is working great with tasmota.  However, after a power loss, then restored, when booting up HA receives invalid data (all value is 0) so I guess the MCU is not ready yet. 

I would like to make a rule to delay the publishing to MQTT data to HA on a first boot, I hope the command is only executed after 60 secs of system boot

I modified the template to

``` 
Rule1 on System#Boot do RuleTimer1 60 endon on Rules#Timer=1 do backlog TuyaSend8; RuleTimer1 10 endon
```

I hope it will delay executing of command TuyaSend8 after 60 secs, then it repeats every 10 secs afterwards. However it seems the command is immediately executed after a system reboot. Perhaps I made mistakes in understanding the rules command?

```
00:00:06.471 QPC: Reset
00:00:07.812 WIF: Connected
00:00:08.058 HTP: Web server active on tasmota_903CB8-7352 with IP address 192.168.8.88
13:58:57.126 RSL: RESULT = {"TuyaReceived":{"Data":"55AA0303000005","Cmnd":3}}
13:58:58.041 MQT: Attempting connection...
13:58:58.106 MQT: Connected
13:58:58.109 MQT: tele/tasmota_903CB8/LWT = Online (retained)
13:58:58.111 MQT: cmnd/tasmota_903CB8/POWER = 
13:58:58.118 MQT: tele/tasmota_903CB8/INFO1 = {"Info1":{"Module":"Tuya MCU","Version":"9.4.0.4(tasmota)","FallbackTopic":"cmnd/DVES_903CB8_fb/","GroupTopic":"cmnd/tasmotas/"}}
13:58:58.126 MQT: tele/tasmota_903CB8/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota_903CB8-7352","IPAddress":"192.168.8.88"}}
13:58:58.135 MQT: tele/tasmota_903CB8/INFO3 = {"Info3":{"RestartReason":"Software/System restart"}}
13:58:58.142 MQT: stat/tasmota_903CB8/RESULT = {"POWER":"ON"}
13:58:58.145 MQT: stat/tasmota_903CB8/POWER = ON
13:58:58.163 RUL: SYSTEM#BOOT performs "RuleTimer1 60"
13:58:58.172 MQT: stat/tasmota_903CB8/RESULT = {"T1":60,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
13:58:58.234 MQT: tele/tasmota_903CB8/RESULT = {"TuyaReceived":{"Data":"55AA0301002A7B2270223A226136646A61627A6D7062627036326777222C2276223A22312E302E30222C226D223A307DE8","Cmnd":1,"CmndData":"7B2270223A226136646A61627A6D7062627036326777222C2276223A22312E302E30222C226D223A307D"}}
13:58:58.240 TYA: MCU Product ID: {"p":"a6djabzmpbbp62gw","v":"1.0.0","m":0}
```

Or do I need to make another rule changing the teleperiod to 60 on systemboot, then change again to 10 secs after 60 secs? How do I accomplish this?

**Comment by sfromis** [comment, marked as answer]
Your brief snippet of the log does not show that the RuleTimer expired or that the TuyaSend8 command was executed. That would be very visible in the style of the already displayed `RUL: SYSTEM#BOOT performs "RuleTimer1 60"`. Note that when a device is configured for Tuya, Tasmota already has the logic to handle initialization, what you do with such rules is only to add additional things, not postpone what is already done. Changing TelePeriod would probably not postpone things either, as Tasmota will publish `tele/` payloads when starting the telemetry reporting, and then TelePeriod is simply the time till next reporting.

**Reply by tesnaa** [reply]
apparently you are right, the TuyaSend8 command did execute after 60 secs of  booting up, then repeats every 10 secs afterwards. 

So the initial tuyareceived topics is sent without the TuyaSend8 command then. Is there a way to somehow delay connecting to MQTT server 60 secs after booting up? The reason I would like to do this because its messed up utility_meter kWh calculation in HA, the data was read at 0 briefly, then back to normal numbers shortly after, making the daily kwh calculation shoots up because HA records 0 to normal number (which is months of usage) in one day.

**Reply by sfromis** [reply]
Steering clear of both HA and TuyaMCU devices, my background for suggestions there is "rather thin" What I'd do in my setup would be to filter away messages not to enter the calculations, but such is highly platform-dependent.
